src/drivers/dshot/DShot.cpp

```diff
@@ -67,6 +67,13 @@ int DShot::init()
 	// Getting initial parameter values
 	update_params();
 
+	// We can't advertise in interrupt context later.
+	_esc_status_pub.advertise();
+	esc_status_s &esc_status = _esc_status_pub.get();
+	esc_status = {};
+	_esc_status_pub.update();
+	up_dshot_set_erpm_callback(&DShot::erpm_trampoline, this);
```

Comment:
Right, or we just publish every time right after we send out the new motor commands. That way, we won't have the very latest result but that shouldn't matter too much as the numbers come in with a delay anyway because we retrieve the results round-robin style.